### PR TITLE
Document Alpha Features

### PR DESCRIPTION
**Pull request by jerop**
# Changes

Today, we have alpha features that are enabled by setting `enable-api-fields` to `"alpha"`. We only have a subset of the alpha features listed in the documentation. 

In this change, we document all the alpha features with references to their TEPs and Releases. If applicable, we add the individual feature flags as well.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/alpha-features.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+| [Hermekton](https://github.com/tektoncd/pipeline/blob/main/docs/hermetic.md)                                                    | [TEP-0025](https://github.com/tektoncd/community/blob/main/teps/0025-hermekton.md)          | [v0.25.0](https://github.com/tektoncd/pipeline/releases/tag/v0.25.0) |                             |
+| [Specifying `onError` for a `Step`](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-onerror-for-a-step) | [TEP-0040](https://github.com/tektoncd/community/blob/main/teps/0040-ignore-step-errors.md) | [v0.27.0](https://github.com/tektoncd/pipeline/releases/tag/v0.27.0) |                             |
+| [Implicit `Parameters`](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md#implicit-parameters)                    | [TEP-0023](https://github.com/tektoncd/community/blob/main/teps/0023-implicit-mapping.md)   | [v0.28.0](https://github.com/tektoncd/pipeline/releases/tag/v0.28.0) |                             |
+
```
Comment (sbwsg) [inline docs/alpha-features.md:29]: PipelineRun Timeouts, Graceful Termination, and Windows Script support are all covered by alpha as well I think.

Relevant doc links:

- [PipelineRun Timeouts](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#configuring-a-failure-timeout)
- [Graceful Termination](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#gracefully-cancelling-a-pipelinerun)
- [Windows Scripts](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#windows-scripts)

🤞 hopefully didn't miss any others - I caught these by searching for "ValidateEnabledAPIFields" in our validation code.

**Comment by pritidesai** [comment]
We have a list of alpha features (subset) documented as part of [install.md](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#alpha-features). I would recommend maintaining such list in a single place and adding a reference to the other if needed.

File: docs/alpha-features.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Alpha Features
+
+- [Introduction](#introduction)
+- [Alpha Table](#alpha-features-table)
+
+## Introduction
+
+This doc provides a list of features in Tekton Pipelines that are considered [Alpha](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#alpha-beta-and-ga).
+
+Set `enable-api-fields` to `"alpha"` to allow alpha features to be used.
```
[the remark this answers is not in the document]
Comment (jerop) [reply docs/alpha-features.md:17]: will do, thanks!

**Comment by jerop** [comment]
> We have a list of alpha features (subset) documented as part of [install.md](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#alpha-features). I would recommend maintaining such list in a single place and adding a reference to the other if needed.

@pritidesai thanks for finding that, will update it with all of them! 😁

**Review by pritidesai — APPROVED** [review]
thanks @jerop

**Pull request merged**